source/include/FreeRTOSIPConfigDefaults.h

```diff
@@ -3119,7 +3122,14 @@
  */
 
 #ifndef FreeRTOS_flush_logging
-    #define FreeRTOS_flush_logging()    if( ipconfigHAS_PRINTF || ipconfigHAS_DEBUG_PRINTF ) do {} while( ipFALSE_BOOL )
+    #define FreeRTOS_flush_logging()                     \
+    if( ipconfigHAS_PRINTF || ipconfigHAS_DEBUG_PRINTF ) \
+    {                                                    \
+        do {} while( ipFALSE_BOOL );                     \
+    }                                                    \
+    else                                                 \
```

Comment:
I don't think we need this else here.

Reply:
Else is what was the MISRA error, it does not allow without an else.

Reply:
I think that coverity doesn't like this statement:
~~~
if( ipconfigHAS_PRINTF || ipconfigHAS_DEBUG_PRINTF )
~~~
because the condition is not variable.

Why not like this:
~~~
#ifndef FreeRTOS_flush_logging
    #if( ipconfigHAS_PRINTF == ipconfigENABLE ) ||
       ( ipconfigHAS_DEBUG_PRINTF == ipconfigENABLE )
        #define FreeRTOS_flush_logging() do {} while( ipFALSE_BOOL );
    #endif
#endif
~~~

Reply:
MISRA Rule 15_6 : The body of the "then" branch of the "if" statement is not a compound statement. This was the error, so with if else is expected. Sure @htibosch let me update according to your suggestion and [PR 1051](https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/1051)